src/Nancy/HttpLinkRelation.cs

```diff
@@ -0,0 +1,180 @@
+﻿namespace Nancy
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// The relation that identifies the semantics of a <see cref="HttpLink"/> contained in an RFC 5899 Link HTTP header,
+    /// as built by the <see cref="HttpLinkBuilder"/>.
+    /// </summary>
+    /// <seealso cref="System.Uri" />
+    [Serializable]
+    public class HttpLinkRelation : Uri, IEquatable<HttpLinkRelation>
```

Comment:
If it is a `Uri`, do we need a separate type for it? As far as I understand, it's basically a `Uri` consisting of a base and relative `Uri`? I think the `Uri` parsing code is a lot more efficient than what's in this type...